src/main/java/org/example/RandomNumberRunnable.java

```diff
@@ -0,0 +1,20 @@
+package org.example;
+
+
+import java.util.Random;
+
+public class RandomNumberRunnable implements Runnable {
+    private int number;
+
+    public int getNumber() {
+        return number;
+    }
+
+    @Override
+    public void run() {
+        Random random = new Random();
+        number = random.nextInt(100000) + 1;
```

Comment:
you can create a constant for this number, your thieves go up to 10.000 but the number is 100.000. creat a final static int MAX_NUMBER=100000 and use it everywhere